src/lib/ui/LoadingToast/LoadingToast.tsx

```diff
@@ -0,0 +1,50 @@
+import classNames from 'classnames'
+import { useEffect, useState } from 'react'
+
+import { CheckBlackIcon } from 'lib/icons/CheckBlack.js'
+import { Spinner } from 'lib/ui/Spinner/Spinner.js'
+
+interface LoadingToastProps {
+  isLoading: boolean
+  top?: number
+  left?: number
+}
+
+export function LoadingToast({
+  isLoading = true,
+  top,
+  left,
+}: LoadingToastProps): JSX.Element {
+  const [hidden, setHidden] = useState(false)
+
+  useEffect(() => {
+    let timeout: ReturnType<typeof globalThis.setTimeout>
+
+    if (!isLoading) {
+      timeout = globalThis.setTimeout(() => {
+        setHidden(true)
+      }, 1000)
+    } else {
+      setHidden(false)
+    }
+
+    return () => {
+      clearTimeout(timeout)
+    }
```

Comment:
Every time I see an `else` I think if it's really needed. In this case I almost believed it's unavoidable, but squinted harder shows it's not. I can't think of a time where removing the else didn't lead to a cleaner pattern or remove a potential bug.


```suggestion
    if (!isLoading) {
      const timeout = globalThis.setTimeout(() => {
        setHidden(true)
      }, 1000)
      return () => {
        globalThis.clearTimeout(timeout)
      }
    }
    setHidden(false)
```

Reply:
[265edd3](https://github.com/seamapi/react/pull/493/commits/265edd36745850a8b4ff2f991c46450be64ef26b)

Reply:
Going to merge this to unblock but we can open another PR if you don't like the no-op